cvat/apps/analytics_report/report/create.py

```diff
@@ -0,0 +1,319 @@
+            rq_job = None
+
+        return rq_job
+
+    def is_custom_analytics_check_job(self, rq_job) -> bool:
+        return rq_job.meta.get("job_type") == self._RQ_CUSTOM_ANALYTICS_CHECK_JOB_TYPE
+
+    @classmethod
+    def _check_job_analytics(cls, *, cvat_job_id: int=None, cvat_task_id: int=None, cvat_project_id: int=None) -> int:
+        if cvat_job_id is not None:
+            db_job = Job.objects.select_related("analytics_report").get(pk=cvat_job_id)
+            return cls()._compute_report_for_job(db_job)
+        elif cvat_task_id is not None:
+            db_task = Task.objects.select_related("analytics_report").prefetch_related("segment_set__job_set").get(pk=cvat_task_id)
+            return cls()._compute_report_for_task(db_task)
+        elif cvat_project_id is not None:
+            db_project = Project.objects.select_related("analytics_report").prefetch_related("tasks__segment_set__job_set").get(pk=cvat_project_id)
+            return cls()._compute_report_for_project(db_project)
+
+    @staticmethod
+    def _get_statistics_entry(statistics_object):
+        return {
+            "title": statistics_object.title,
+            "description": statistics_object.description,
+            "granularity": statistics_object.granularity,
+            "default_view": statistics_object.default_view,
+            "transformations": statistics_object.transformations,
+            "dataseries": statistics_object.calculate(),
+        }
+
+    def _compute_report_for_job(self, db_job):
+        db_report = getattr(db_job, "analytics_report", None)
+        was_created = False
+        if db_report is None:
+            db_report = AnalyticsReport.objects.create(
+                job_id=db_job.id,
+                statistics={})
+            db_report.save()
+            was_created = True
+            db_job.analytics_report = db_report
+
+        # recalculate the report if it is not relevant
+        if db_report.created_date < db_job.updated_date or was_created:
+
+            annotation_speed = JobAnnotationSpeed(db_job)
+            objects = JobObjects(db_job)
+            annotation_time = JobAnnotationTime(db_job)
+
+            statistics = {
+                "objects": self._get_statistics_entry(objects),
+                "annotation_speed": self._get_statistics_entry(annotation_speed),
+                "annotation_time": self._get_statistics_entry(annotation_time),
+            }
+
+            total_annotation_speed = JobTotalAnnotationSpeed(db_job, primary_statistics=statistics["annotation_speed"])
+            total_object_count = JobTotalObjectCount(db_job, primary_statistics=statistics["annotation_speed"])
+
+            statistics["total_annotation_speed"] = self._get_statistics_entry(total_annotation_speed)
+            statistics["total_object_count"] = self._get_statistics_entry(total_object_count)
+
+            db_report.statistics = statistics
+            db_report.save()
+
+        return db_report
```

Comment:
Probably, this function should return an integer id or other simple object (or maybe it should do the calling function instead). RQ will need to serialize the object (likely via pickle) to save the result.